### PR TITLE
Achievements

### PR DESCRIPTION
**Pull request by maksgritchin**

### CODE REVIEW

File: OutOfSchool/OutOfSchool.WebApi/Controllers/V1/AchievementController.cs [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+{
+    /// <summary>
+    /// Controller with CRUD operations for Achievement entity.
+    /// </summary>
+    [ApiController]
+    [ApiVersion("1.0")]
+    [Route("api/v{version:apiVersion}/[controller]/[action]")]
+    public class AchievementController : ControllerBase
+    {
+        private readonly IAchievementService service;
```
Comment (valerabad) [inline OutOfSchool/OutOfSchool.WebApi/Controllers/V1/AchievementController.cs:26]: Better use achievementService name instead service, I think

File: OutOfSchool/OutOfSchool.WebApi/Services/AchievementService.cs [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+                var achievement = await repository.GetById(dto.Id).ConfigureAwait(false);
+
+                achievement.AchievementTypeId = dto.AchievementTypeId;
+                achievement.Title = dto.Title;
+                achievement.AchievementDate = dto.AchievementDate;
+                achievement.WorkshopId = dto.WorkshopId;
+                achievement.Children.RemoveAll(x => dto.ChildrenIDs.IndexOf(x.Id) < 0);
+                achievement.Children.AddRange(context.Children.Where(w => dto.ChildrenIDs.Contains(w.Id)).ToList());
+
+                achievement.Teachers.RemoveAll(x => dto.Teachers.IndexOf(x.Title) < 0);
```
Comment (provicevko) [inline OutOfSchool/OutOfSchool.WebApi/Services/AchievementService.cs:130]: `achievement.Teachers.RemoveAll(x => !dto.Teachers.Contains(x.Title));`

File: OutOfSchool/OutOfSchool.WebApi/Services/AchievementService.cs [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+                foreach (var Teacher in dto.Teachers)
+                {
+                    if (achievement.Teachers.Find(x => x.Title == Teacher) is null)
```
Comment (provicevko) [inline OutOfSchool/OutOfSchool.WebApi/Services/AchievementService.cs:133]: You could rewrite something like this.
```
var exceptTeachers = dto.Teachers
   .Where(p => achievement.Teachers.All(x => !x.Title.Equals(p)));
foreach (var teacherId in exceptTeachers)
{
   achievement.Teachers.Add(new AchievementTeacher { Title = dto.Title, Achievement = achievement });
}
```
**Should compare strings by `Equals()` method**

**Comment by provicevko** [comment]
And also I haven't seen your new migration considering new Db sets (if you remember, it's Ok :))

**Pull request merged**